Build: Tests: "make test" now fails if there are failing or broken tests

inkarkat opened the pull request:
`aggregate-results.sh` just reports the execution status, but succeeds in all cases.
The build should fail if there are errors. (This also means that no cleanup runs and the failures can be investigated in more detail.)


**Before submitting a pull request,** please make sure the following is done:

- [x] Fork [the repository](https://github.com/todotxt/todo.txt-cli) and create your branch from `master`.
- [ ] If you've added code that should be tested, add tests!
- [x] Ensure the test suite passes.
- [x] Lint your code with [ShellCheck](https://www.shellcheck.net/).
- [ ] Include a human-readable description of what the pull request is trying to accomplish.
- [x] Steps for the reviewer(s) on how they can manually QA the changes: Make a test fail, run `$ make test`, observe exit status 1
- [ ] Have a `fixes #XX` reference to the issue that this pull request fixes.

inkarkat committed:
Build: Tests: "make test" now fails if there are failing or broken tests

aggregate-results.sh just reports the execution status, but succeeds in all cases.
The build should fail if there are errors. (This also means that no cleanup runs and the failures can be investigated in more detail.)

hyperupcall commented:
Kind of surprised that ShellCheck didn't catch the use of `-a` within `[`. Otherwise looks good to me

inkarkat committed:
aggregate-results: Style: Address shellcheck warnings

inkarkat committed:
aggregate-results: Cleanup: Remove unused $quiet

It's not parsed here (from --quiet, as in test-lib.sh), and even exporting it doesn't have any effect, as it's only used when say_color()'s first argument is invalid / empty, which cannot happen due to the defaulting in get_color().

inkarkat commented:
@hyperupcall Thanks for reviewing! I have an older shellcheck 0.7.0 on the system that didn't catch the `-a`. I've fixed that and the all other warnings in that file now.

inkarkat committed:
Tests: Cleanup: Remove unused test_external[_without_stderr]()

These make sense in the original context of testing Git commands (that may have external test scripts), but for todo.sh we've never used them, nor are they likely to be useful for this project.
